notebooks/utils/model_upcast_utils.py

```diff
@@ -0,0 +1,193 @@
+    parameters = []
+    for input_val in input_vals:
+        parameters.append(Parameter(get_element_type(input_val.dtype), ov.PartialShape(input_val.shape)))
+
+    if op.get_type_name() not in ops_to_track_map.keys():
+        # todo: implement for other ops
+        raise NotImplementedError(f"inference track for operations {op.get_type_name()} are not implemented yet")
+
+    new_op = ops_to_track_map[op.get_type_name()](*parameters, **op.get_attributes())
+    ov_model = ov.Model([new_op], parameters)
+
+    exec_net = ov.Core().compile_model(ov_model, 'GPU', config={"INFERENCE_PRECISION_HINT": precision})
+    request = exec_net.create_infer_request()
+    result = request.infer(input_vals)
+    assert len(result) == 1
+    del request, exec_net, ov_model
+    return result[0]
+
+
+def is_model_partially_upcasted(model) -> bool:
+    for node in model.get_ordered_ops():
+        if node.get_type_name() not in ops_to_track_map.keys():
+            continue
+        if 'disable_fp16_compression_0' in node.get_rt_info().keys():
+            return True
+    return False
+
+
+def mark_nodes_to_upcast_to_fp32(model: ov.Model, nodes: List[Node], fp16_infer_vals: List, fp32_infer_vals: List,
+                                 thresholds: None) -> None:
+    nodes_with_errors = []
+    for node, fp16_val, fp32_val in zip(nodes, fp16_infer_vals, fp32_infer_vals):
+        if compare_tensors(node, fp16_val, fp32_val, thresholds):
+            nodes_with_errors.append(node.get_friendly_name())
+
+    for node in model.get_ordered_ops():
+        if node.get_friendly_name() in nodes_with_errors:
+            node.get_rt_info()['disable_fp16_compression_0'] = ''
+
+
+def compare_tensors(node: Node, a: np.ndarray, b: np.ndarray, new_thresholds_per_op, silent: bool = True) -> bool:
+    """
+    If values differ more than a certain metric then function returns True
+    """
+    assert np.array_equal(a.shape, b.shape), f'Shapes differ {a.shape} and {b.shape}'
+    out_size = int(np.prod(a.shape))
+    a_, b_ = np.reshape(a, out_size), np.reshape(b, out_size)
+
+    import warnings
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        rel_error = np.abs(2 * (a_ - b_) / (np.abs(a_) + abs(b_)))
+
+    mean_rel_error = np.mean(rel_error)
+    thresholds_map = get_thresholds_per_op()
+    if new_thresholds_per_op is not None:
+        thresholds_map.update(new_thresholds_per_op)
+    thresholds = thresholds_map[node.get_type_name()]
+    rel_threshold = thresholds[0]
+    rel_threshold_ratio = thresholds[1]
+    rel_tol = thresholds[2]
+
+    rel_diff_ratio = np.size(np.where(rel_error >= rel_threshold)) / out_size
+    if mean_rel_error < rel_tol:
+        return False
+    if rel_diff_ratio > rel_threshold_ratio:
+        if not silent:
+            print(f'upcasted node {node.get_friendly_name()} with 0.1 rel2_diff_ratio {rel_diff_ratio} and mean_rel_error {mean_rel_error}')
+        return True
```

Comment:
Could you check it with flake8? There no new line at the end of file and a couple of lines are longer then 120 characters

Reply:
i suggest to apply it in a separate PR